### PR TITLE
Add support for Cupertino widgets and all the adaptive widgets

### PR DESCRIPTION
**Pull request by freshtechtips**
I checked one of the issues in [flutter_quill](https://pub.dev/packages/flutter_quill) and he/she was having an issue when using [flutter_quill](https://pub.dev/packages/flutter_quill) in the **FluentApp()**, when I tried to encounter that issue, and it was not directly related to the Flutter QuillEditor however I have encountered another issue when using adaptive widgets in **FluentApp** like **AdaptiveTextSelectionToolbar()** which is official from flutter team, we will get crash since **Cupertino localizations** is required and **Fluent app** doesn't include it so the devs have to override the **localizationsDelegates** with the following

```dart

  localizationsDelegates: const [
        DefaultCupertinoLocalizations.delegate,
        DefaultMaterialLocalizations.delegate,
        DefaultWidgetsLocalizations.delegate,
        FluentLocalizations.delegate,
      ],

```

when using FluentApp.

Before:

image

After:

image

I also changed the default scroll bar for the **Axis.vertical** on macOS and iOS to CupertinoScrollBar which looks better, in my opinion, all the users who are used to using Apple products, of course you can reject these changes but the DefaultCupertinoLocalizations is already included by default in flutter_localizations and it's imported already so I don't see if there is a reason to not using it, that is up to the maintainers of this package

Suggestion:
Why we are not using the ScrollBar (material version) widget as default for **Android** and **Fuchsia OS**??

## Pre-launch Checklist

- [x] I have updated `CHANGELOG.md` with my changes  
- [x] I have run "dart format ." on the project  
- [x] I have added/updated relevant documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 4.7.7
+- Add a support for Cupertino Loclizations from GlobalCupertinoLocalizations, this can help fix some errors when using offical adaptive widgets and other cupertino widgets
+- Now, By default we will use CupertinoScrollbar() widget for iOS and macOS, you can change this in the scrollBehavior to fit your needs if needed.
```
Comment (bdlukaa) [inline CHANGELOG.md:3]: Add this under the `[next]` section. The next published version will contain the changes.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: fluent_ui
 description: Implements Windows UI in Flutter. Based on the official documentation
-version: 4.7.6
+version: 4.7.7
```
Comment (bdlukaa) [inline pubspec.yaml:3]: Do not change the version.

**Review by bdlukaa — CHANGES_REQUESTED** [review]
You may need to rebase

**Comment by freshtechtips** [comment]
> You may need to rebase



> You may need to rebase

Please take a look at [scroll_pos](https://pub.dev/packages/scroll_pos/changelog) change log
it has a breaking change
so I updated it to the latest version

**Comment by freshtechtips** [comment]
Hi again, You might want to consider replying to this suggestion if possible:
> Why not use the ScrollBar (material version) widget as default for Android and Fuchsia OS??

**Comment by bdlukaa** [comment]
> Please take a look at scroll_pos change log
it has a breaking change

#947 has fixed this. Please revert

**Comment by freshtechtips** [comment]
> > Please take a look at scroll_pos change log
> > it has a breaking change
> 
> #947 has fixed this. Please revert

Done

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,7 +24,7 @@
 * fix: expose more `TextField` properties on `PasswordBox` ([#925](https://github.com/bdlukaa/fluent_ui/issues/925))
 * fix: `AutoSuggestBox.onOverlayVisibilityChanged` now results in the correct state ([#926](https://github.com/bdlukaa/fluent_ui/issues/926))
 * fix: `MenuFlyoutSubItem` inherits the acrylic data from `MenuFlyout` ([#932](https://github.com/bdlukaa/fluent_ui/issues/932))
-* fix: `MenuFlyoutSubItem` inhertis its `MenuFlyout` parent decoration data ([#931](https://github.com/bdlukaa/fluent_ui/issues/931)) 
+* fix: `MenuFlyoutSubItem` inhertis its `MenuFlyout` parent decoration data ([#931](https://github.com/bdlukaa/fluent_ui/issues/931))
```
Comment (bdlukaa) [inline CHANGELOG.md:27]: Why is the whole file changed?

Comment (freshtechtips) [reply CHANGELOG.md:27]: Auto formatting by vs code extension, I don't think it will cause any damage

Comment (freshtechtips) [reply CHANGELOG.md:27]: > Why is the whole file changed?

Hi, do you want me to revert this change??

Comment (bdlukaa) [reply CHANGELOG.md:27]: Yes, please!

Comment (freshtechtips) [reply CHANGELOG.md]: > Yes, please!

Done

**Review by bdlukaa — APPROVED** [review]

**Pull request merged**